### ISSUE TITLE
Backspace bad behaviour

### ISSUE THREAD

**Issue by jbarren** [tmux/tmux#321]
Backspace doesn't behave as it should. If two characters are typed, and backspace is pressed let's say ten times, every backspace press will print a space, but the limit of the spaces printed will be two and the error bell will ring in the rest of the eight times. If five characters are typed the limit will be five and so on. 
Setup:
- Tmux version: 2.1-1
- Cygwin 64 on Windows 7 x64
- Mintty 2.2.3

Tried with different terminal models in Mintty (xterm, xterm-256, vt220...) but didn't make a difference: the behavior of the key is correct in the terminal until I enter in a tmux session. 
Ctrl-H nor Ctrl-V behave like backspace, and tried enabling the option to send Ctrl-H in every backspace in Mintty options, but didn't make a difference either. 

Current tmux.conf:
```
#set -g default-terminal "xterm-256-color"

# remap prefix from 'C-b' to 'C-a'
unbind C-b
set-option -g prefix C-a
bind-key C-a send-prefix

# split panes using | and -
bind | split-window -h
bind - split-window -v
unbind '"'
unbind %

# switch panes using Alt-arrow without prefix
bind -n M-h select-pane -L
bind -n M-l select-pane -R
bind -n M-k select-pane -U
bind -n M-j select-pane -D

# Enable mouse control (clickable windows, panes, resizable panes)
set -g mouse on
```

**Comment by nicm** [comment]
Backspace in tmux sends ^? not ^H. This sounds like a a problem with
your shell to me.

**Comment by jbarren** [comment]
Ok, I haven't tried with ^?, but you might be right. What other tests can I
make to discard my shell? I'm afraid that I won't be able to test it until
the 14th though (it's the work pc, and I'm off until then)

**Comment by nicm** [comment]
You could try to configure your terminal to send ^? for backspace too.

**Comment by cayossarian** [comment]
I had the same problem (with backspace plus getting addition characters from auto completion) whenever I entered a tmux session.  I removed the .tmux.conf and the problem disappeared so I changed the following in my .tmux.conf

changed from:
set -g default-terminal "tmux-256color"
to:
set -g default-terminal "xterm-256color"

This solved the problem.  Removing this line altogether also worked but it seemed better to be explicit.
